Multisig/UI/Settings/App Settings/Owner Wallet Management/SelectOwnerAddressView.swift

```diff
@@ -10,22 +10,26 @@ import SwiftUI
 import Web3
 
 struct SelectOwnerAddressView: View {
-    @Binding
-    var rootIsActive: Bool
-
+    @Environment(\.presentationMode)
+    var presentationMode: Binding<PresentationMode>
+    
     private var rootNode: HDNode?
-
+    var onSubmit: (() -> Void)?
+    
     @State
     private var addresses = [Address]()
 
     @State
     private var selected = 0
 
-    init(rootNode: HDNode?, rootIsActive: Binding<Bool>) {
+    private var maxAddressesCount = 100
+    private var pageSize = 20
+
+    init(rootNode: HDNode?, onSubmit: (() -> Void)? = nil ) {
         self.rootNode = rootNode
-        _rootIsActive = rootIsActive
-        let addresses: [Address] = (0..<20).compactMap { self.addressAt($0) }
+        let addresses: [Address] = generateAddressesPage()
         _addresses = State(initialValue: addresses)
+        self.onSubmit = onSubmit
     }
 
     private func addressAt(_ index: Int) -> Address? {
```

Comment:
- this should be in a view model
- the errors here are not handled - will crash if the private key from the data fails (line 39)

Reply:
@DmitryBespalov There is no view model defined for this screen at all, should we define it in this ticket ?

Reply:
could be in the next pr 👍